### PR TITLE
hostapd: Update hostapd src to 2021-04-25

### PR DESCRIPTION
**Pull request by mgiganto**
Update package, removed patches already integrated in src and update
patches to the new src.

Signed-off-by: Manuel Giganto <mgigantoregistros@gmail.com>

### CODE REVIEW

File: package/network/services/hostapd/Makefile [hunk 1]
```diff
@@ -7,13 +7,13 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=hostapd
-PKG_RELEASE:=32
+PKG_RELEASE:=33
```
Comment (dangowrt) [inline package/network/services/hostapd/Makefile:10]: As `PKG_SOURCE_DATE` is bumped you should reset `PKG_RELEASE` to `1` or (preferably) use `$(AUTORELEASE)`

**Review by dangowrt — CHANGES_REQUESTED** [review]

File: package/network/services/hostapd/Makefile (hunk 1 above)
Comment (mgiganto) [reply package/network/services/hostapd/Makefile]: Thank you @dangowrt!
PKG_RELEASE:=$(AUTORELEASE) it is!
This is changed now.
Let me know if you see something else.

**Comment by hauke** [comment]
I also looked into this and I had to backport this patch in addition:
https://git.kernel.org/linus/9f0ffa418483938d25a15f6ad3891389f333bc59
Without this patch NL80211_ATTR_RECONNECT_REQUESTED had the wrong offset and failed when called from hostapd

**Comment by mgiganto** [comment]
> I also looked into this and I had to backport this patch in addition:
> https://git.kernel.org/linus/9f0ffa418483938d25a15f6ad3891389f333bc59
> Without this patch NL80211_ATTR_RECONNECT_REQUESTED had the wrong offset and failed when called from hostapd
Thank you @hauke, how do you suggest to incorporate that patch to the kernel?
Include the kernel patch in this PR to both kernels?

**Comment by hauke** [comment]
I pushed my work in progress patches for hostapd here:
https://git.openwrt.org/?p=openwrt/staging/hauke.git;a=shortlog;h=refs/heads/hostapd

**Comment by mgiganto** [comment]
@hauke I have included your mac80211 patch to this PR

**Comment by huaracheguarache** [comment]
I've been running a build with this PR for almost 12 hours now without any issues. In my setup I have a FT-SAE network on 5 GHz, and WPA3/WPA2-mixed network on 2.4 GHz.

**Comment by mgiganto** [comment]
I have detected some sort of problem for some devices like galaxy s7 when conneting using EAP.
I am investigating, but it seems related with 
`
67efd19e0a6019c11fd83a47f68a3573da1db520 
From: Markus Theil <markus.theil@tu-ilmenau.de>
Date: Wed, 10 Jun 2020 10:32:58 +0200
Subject: [PATCH 071/871] nl80211: Use control port TX (status) in AP mode if
 possible

Check if nl80211 control port TX status is available in the kernel and
enable control port TX if so. With this feature, nl80211 control path is
able to provide the same feature set as nl80211 (management) + AF_PACKET
socket (control) before.
`
I am investigating the issue

**Comment by mgiganto** [comment]
I have created a new patch that prevent the activation of control_port_ap from 67efd19e0a6019c11fd83a47f68a3573da1db520 this prevents the EAP timeout until the issue is resolved:

> IEEE 802.1X: Sending EAP Packet (identifier 113)
> WPA: sending 1/4 msg of 4-Way Handshake
> WPA: received EAPOL-Key frame (2/4 Pairwise)
> WPA: sending 3/4 msg of 4-Way Handshake
> WPA: EAPOL-Key timeout
> WPA: sending 3/4 msg of 4-Way Handshake
> WPA: EAPOL-Key timeout
> WPA: sending 3/4 msg of 4-Way Handshake
> WPA: EAPOL-Key timeout
> WPA: sending 3/4 msg of 4-Way Handshake
> WPA: EAPOL-Key timeout
> WPA: PTKINITNEGOTIATING: Retry limit 4 reached

**Comment by mgiganto** [comment]
Removed the 300 and 500 patch from @hauke's mac80211: Fix patch, it is now part of master.

**Comment by mgiganto** [comment]
Patch updated. It compiles again on master (6713fe030fca32fc3d5ad9761f3b2f96501aedd6)
@huaracheguarache please try again with the updated version

**Comment by huaracheguarache** [comment]
I can confirm that master builds successfully with this PR now. Nice work! =)

**Comment by mgiganto** [comment]
@Felix Fietkau has commit an updated version (2021-05-21) here https://git.openwrt.org/?p=openwrt/staging/nbd.git;a=commit;h=efee3bc92cce1ccc3bf24f8b3fdb9137054dc908
I am closing this PR

**Pull request closed without merging**